eventuals/expected.h

```diff
@@ -28,6 +29,7 @@ auto ExpectedToEventual(tl::expected<T, E>&& expected) {
       .template raises<
           std::conditional_t<
               std::disjunction_v<
+                  CheckErrorsTypesForVariant<std::decay_t<E>>,
                   std::is_base_of<std::exception, std::decay_t<E>>,
                   std::is_base_of<std::exception_ptr, std::decay_t<E>>>,
               E,
```

Comment:
I never want to "raise" a `std::variant` for now, so instead, let's extract out all of the error types. This is hard because `Eventual().raises<E1, E2>()` doesn't take a `std::tuple`, so to simplify your life how about we provide the following overload `ExpectedToEventual(tl::expected<T, std::variant<Errors...>>&& expected)` and then deal with the `std::variant` case explicitly!